storage/tree_storage.go

```diff
@@ -19,6 +19,11 @@ type TreeTX interface {
 
 	// Rollback aborts any performed operations. No updates must be applied to the underlying storage.
 	Rollback() error
+
+	// Open indicates if this transaction is open. An open transaction is one for which
+	// Commit() or Rollback() has never been called. Implementations must do all clean up
+	// in these methods so transactions are assumed closed regardless of the reported success.
+	Open() bool
```

Comment:
I'd much prefer IsOpen but Get in Getters isn't idiomatic apparently. I'm going to make it IsOpen anyway.